### DISCUSSION TITLE
Dates do not survive 'import chats'

### DISCUSSION THREAD

**Discussion by mrmvn** [open-webui/open-webui#5682] [Ideas]
**Is your feature request related to a problem? Please describe.**
Created and updated dates for each chat do not seem be imported during chat imports.

**Describe the solution you'd like**
I would like this information to be kept when importing chats, i.e. not replaced by the very date/time when the old chats were imported. 
For sorting chats on the UI, I guess the one that matters most is the _updated date_, so we could in principle replace the _created date_ with the import date with less impact, though I personally don't see how this could be useful so I'm not recomending anything in particular there.

**Describe alternatives you've considered**
Considered to mess around with the existing database with imported chats to change the dates based on .json file to import, but thought it'd probably be easier to actually implement the feature if desirable for others too.

**Comment by agibson-fl** [comment]
I'm experiencing this too. I thought it was related to importing chats from chatgpt and that feature, but I just tested exporting and importing from one open-webui instance to another. Dates aren't set on import in either case.

Here's where I commented on another discussion thinking (incorrectly) that it was related to the feature/code changes to import chats from chatgpt: https://github.com/open-webui/open-webui/discussions/5205#discussioncomment-10976338

**Reply by mahenning** [reply]
I found out about this today. Would be really nice if this is not happening. I looked nto the .json file of the exported chat, and the `timestamp`, `created_at` and `updated_at` values are correct. When exporting the newly imported chat (the one with "today" now) again, the `created_at` and `updated_at` now has the time code of the import time. I assume it's because the chat is literally created and then flled with the logs, but it would be nice to have a better import.

**Comment by samsong** [comment]
manual fix possible. time consuming but worth it I guess.

- Export ChatGPT json.

- Open file: regex remove the seconds from timestamp, save close

- Upload file to: https://tableconvert.com/json-to-sql - download "insert sql" (sqlite) including table creation, tablename "chatgpt"

- Download openwebui database: filename webui.db 

- Open it using (Windows) sqlite db browser

- Import chat SQL file to a table called "chatgpt"

- run this command:
```
UPDATE chat
SET created_at = (
    SELECT chatgpt.create_time 
    FROM chatgpt 
    WHERE chatgpt.title = chat.title
),
updated_at = (
    SELECT chatgpt.update_time 
    FROM chatgpt 
    WHERE chatgpt.title = chat.title
)
WHERE EXISTS (
    SELECT 1 FROM chatgpt WHERE chatgpt.title = chat.title
);
```

- close the database, upload file replace webui.db

- run openwebui. all chats will be date sorted.

![owebui](https://github.com/user-attachments/assets/b704591a-54b6-439b-9ca2-f691d409fcce)

**Comment by FarisHijazi** [comment]
I found the problem, the import code has a tiny bug, I fixed it here, awaiting the PR to be merged: https://github.com/open-webui/open-webui/pull/11808

**Comment by NodeJSmith** [comment]
@FarisHijazi or @mariomorvan - would either of you be able to explain to me how this would work through the UI, once the PR is merged into main? I see the manual steps, but this implies there is a less manual process, but I cannot find any documentation or anything in the settings that looks like it would facilitate importing ChatGPT history

**Reply by rgaricano** [reply]
You won't have to do anything, it's a correction of the import process (which doesn't update the dates correctly when the import is performed).
 export/import Chats are in User-Settings-Chat
